src/cli/trn_cli_common.c

```diff
@@ -151,6 +153,27 @@ int trn_cli_parse_net(const cJSON *jsonobj, struct rpc_trn_network_t *net)
 		}
 	}
 	net->switches_ips.switches_ips_len = i;
+	// Cluster gateway is NULL or empty if it is not set in a config map which is a string value
+	net->cluster_gateway = 0;
+	if (cluster_gateway != NULL && cJSON_IsString(cluster_gateway) && strlen(cluster_gateway->valuestring) > 0) {
+		struct sockaddr_in sa;
+		inet_pton(AF_INET, cluster_gateway->valuestring, &(sa.sin_addr));
+		net->cluster_gateway = sa.sin_addr.s_addr;
+	}
+        // Virtual can be NULL if it is a local subnet. 0 as false and 1 as true
+	net->virtual = 0;
```

Comment:
why 0 is used instead of true?

Reply:
There is no bool type in the linux kernel.

Reply:
got it.